components/expand-collapse/expand-collapse.js

```diff
@@ -0,0 +1,132 @@
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+import ResizeObserver from 'resize-observer-polyfill/dist/ResizeObserver.es.js';
+import { styleMap } from 'lit-html/directives/style-map.js';
+
+const reduceMotion = matchMedia('(prefers-reduced-motion: reduce)').matches;
+
+const states = {
+	COLLAPSE: 'collapse', // in the process of collapsing or fully collapsed
+	EXPANDING: 'expanding', // in the process of expanding
+	EXPANDED: 'expanded', // fully expanded
+};
+
+class ExpandCollapse extends LitElement {
+
+	static get properties() {
+		return {
+			expanded: { type: Boolean, reflect: true },
+			_height: { type: Number },
+			_state: { type: String }
+		};
+	}
+
+	static get styles() {
+		return css`
+			:host {
+				display: block;
+			}
+
+			:host([hidden]) {
+				display: none;
+			}
+
+			.d2l-expand-collapse-container {
+				height: 0px;
+				overflow: hidden;
+				transition: height 400ms cubic-bezier(0, 0.7, 0.5, 1);
+			}
+
+			.d2l-expand-collapse-container[data-state="expanded"] {
+				overflow: visible;
+			}
+
+			/* prevent margin colapse on slotted children */
+			.d2l-expand-collapse-content:before,
+			.d2l-expand-collapse-content:after {
+				content: ' ';
+				display: table;
+			}
+
+			@media (prefers-reduced-motion: reduce) {
+				.d2l-expand-collapse-container {
+					transition: none;
+				}
+			}
+		`;
+	}
+
+	constructor() {
+		super();
+		this._onContentResize = this._onContentResize.bind(this);
+		this.expanded = false;
+		this._state = states.COLLAPSE;
+	}
+
+	get expanded() {
+		return this._expanded;
+	}
+
+	set expanded(val) {
+		const oldVal = this._expanded;
+		if (oldVal !== val) {
+			this._expanded = val;
+			this.requestUpdate('expanded', oldVal);
+			this._expandedChanged(val);
```

Comment:
Could we test out our next theory that if we're using reflected attributes that we can just rely on the `attributeChangedCallback` to be notified of changes (and avoid setters/getters)?

Reply:
`attributeChangedCallback` does get called but it doesn't appear to be correctly triggering a re-render when `_state` and `_height` get set. 😕

Reply:
Perhaps [this bug](https://github.com/Polymer/lit-element/issues/965).